### PR TITLE
test(fonts): unit test unicode range parsing

### LINKED ISSUE #159: inRange parses Unicode ranges inside the offline e2e spec, untested
Raised by review on #158 (`tests/e2e/offline.spec.ts`).

`inRange` parses a `unicodeRange` string — splitting on commas, stripping a
`u+` prefix, handling a single code point as well as a `low-high` pair, and
`parseInt(..., 16)` on each end. It is pure logic, it lives inside a Playwright
spec, and nothing tests it.

It decides whether the assertion in
`draws the headword in its own face, which is the thing the reader came for`
passes. A parsing bug there is invisible in both directions: read too narrowly
it fails a correct build, read too broadly it passes a broken one, and that
second case is the one the test exists to catch.

The remedy is a restructure rather than a move. `inRange` runs inside
`page.evaluate`, so it cannot simply be imported — the browser context does not
close over Node's modules. The shape that works is to have `page.evaluate`
return the raw `unicodeRange` strings of the loaded faces, which are
serialisable, and do the parsing in Node with a helper the unit suite covers.
That also matches what `CLAUDE.md` asks of the two layers: the browser reports
what it loaded, and the pure interpretation of that data is a unit test.

Not done on the release branch because it rewrites a working test's structure
between a release commit and production.

Note the review's other suggestion, `document.fonts.check()`, is not the answer
and the reviewer said so itself: it returns `true` when the text can be rendered
without a network load, which includes falling back to a system font. It would
report success in exactly the case this test is here to fail.

### PR DESCRIPTION
**Pull request by kowinauyeung**
### Summary

Closes #159.

Move CSS unicode-range interpretation out of the offline Playwright browser context so the parser is independently covered.

### Changes

- Add unicodeRangeIncludes for comma-separated CSS ranges, U+ prefixes, single code points, and inclusive bounds.
- Return serializable unicodeRange strings from document.fonts in the offline spec and interpret them in Node.

### Testing

Unit tests are the selected layer because Unicode range parsing is pure input-to-output logic. The existing offline E2E remains responsible for verifying which font faces the browser loads through the service worker.

- yarn test:unit
- yarn typecheck
- yarn playwright test tests/e2e/offline.spec.ts
- yarn lint (passes with 19 pre-existing warnings)
- yarn format and yarn format:check

Regression proof: temporarily changed the inclusive upper-bound check to an exclusive one. Only the singleton and upper-bound cases in tests/unit/unicodeRange.test.ts failed, then passed again after restoring the inclusive check.

## Summary by CodeRabbit

* **New Features**
  * Added support for checking whether Unicode code points are included in CSS Unicode ranges.
  * Supports comma-separated ranges, optional case-insensitive `U+` prefixes, single code points, and inclusive boundaries.

* **Tests**
  * Added coverage for Unicode range parsing and boundary cases.
  * Updated font-loading checks to use the shared Unicode range validation.

### CODE REVIEW

File: src/lib/unicodeRange.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/** Whether a CSS unicode-range includes a Unicode code point. */
+export function unicodeRangeIncludes(range: string, codePoint: number): boolean {
+  return range.split(',').some((part) => {
+    const [start, end] = part.trim().replace(/^u\+/i, '').split('-');
+    const low = Number.parseInt(start ?? '', 16);
+    const high = end === undefined ? low : Number.parseInt(end, 16);
```
[the remark this answers is not in the document]
Comment (kowinauyeung) [reply src/lib/unicodeRange.ts]: @qodo-code-review
Fixed. unicodeRangeIncludes now expands each wildcard to 0 for the lower bound and F for the upper bound, so U+4?? covers U+0400 through U+04FF. Added inclusive lower- and upper-bound regression cases.
Fixed in 1421509
